server/src/domain/storage/storage.core.ts

```diff
@@ -143,26 +168,93 @@ export class StorageCore {
       this.logger.log(`Attempting to finish incomplete move: ${move.oldPath} => ${move.newPath}`);
       const oldPathExists = await this.repository.checkFileExists(move.oldPath);
       const newPathExists = await this.repository.checkFileExists(move.newPath);
-      const actualPath = newPathExists ? move.newPath : oldPathExists ? move.oldPath : null;
+      const actualPath = oldPathExists ? move.oldPath : newPathExists ? move.newPath : null;
       if (!actualPath) {
         this.logger.warn('Unable to complete move. File does not exist at either location.');
         return;
       }
 
-      this.logger.log(`Found file at ${actualPath === move.oldPath ? 'old' : 'new'} location`);
+      const fileAtNewLocation = actualPath === move.newPath;
+      this.logger.log(`Found file at ${fileAtNewLocation ? 'new' : 'old'} location`);
+
+      if (fileAtNewLocation) {
+        if (!(await this.verifyNewPathContentsMatchesExpected(move.oldPath, move.newPath, assetInfo))) {
+          this.logger.fatal(
+            `Skipping move as file verification failed, old file is missing and new file is different to what was expected`,
+          );
```

Comment:
Is there any more action we can take in this case? Do assets that hit here implicitly also show up as offline on the repair page, or do we want to do something to make that happen?

Reply:
I think at this point you're pretty much screwed. Even with manual intervention I don't think you can do anything to fix this given that the original file is lost. Notifying the user that a file got lost and that they restore it from a backup rather quickly seems like a good idea in general. Not sure if the repair page is a good place for this though. People may not look at it (at all).

Reply:
At this point the old file is somehow gone and the new file is corrupt, so you're f**ked. That's why I log it as fatal. The file will show up as orphaned. It's worth noting that with all these extra checks, it should actually be impossible to get to this state now.